source/priority_queue.c

```diff
@@ -0,0 +1,133 @@
+/*
+* Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/priority_queue.h>
+#include <string.h>
+
+#define parent_of(index) (index & 1 ? index >> 1 : index > 1 ? (index - 2) >> 1 : 0)
+#define left_of(index) ((index << 1) + 1)
+
+/* Precondition: with the exception of the first element, the container must be in heap order */
+static void sift_down(struct aws_priority_queue *queue) {
+    size_t root = 0;
+    size_t left = left_of(root);
+    size_t len = aws_array_list_length(&queue->container);
+    void *right_item, *left_item, *root_item;
+
+    while(left < len) {
+        aws_array_list_get_at_ptr(&queue->container, &left_item, left);
+
+        size_t right = left + 1;
+        if(right < len) {
+            aws_array_list_get_at_ptr(&queue->container, &right_item, right);
+
+            /* choose the larger/smaller of the two in case of a max/min heap respectively */
+            if (queue->pred(left_item, right_item)) {
+                left = right;
```

Comment:
Don't call it left if it's not left.

Reply:
meh, it's a convenience, having left, right and swap made the code less readable.